### DISCUSSION TITLE
Is it possible to tell renovate that a reusable workflow has been updated in a non backwards compatible manner?

### DISCUSSION THREAD

**Discussion by electrofelix** [renovatebot/renovate#26171] [Request Help]
### How are you running Renovate?

None

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Currently evaluating how to manage automatically updating dependencies as well as github actions/workflows.

Have a number of reusable workflows and have previously encountered an issue where someone has written such a workflow that is in use, but doing something that is incorrect. Perhaps it's building and deploying as single job, when it really should only be doing a build and there is another workflow that should be called to handle the deploy.

The problem is that often it is easier to use digest pinning for such workflows from the consuming as often a few will be located in the same repository to be used across multiple projects. While the pinning allows for easy reverting if the updated workflow doesn't work, it's often not obvious until a project is moved forward, and in the case of a deploy being separated off to a separate job, it might not be obvious to some consumers for a while, because it has broken.

It would be really handy if there was some way in the reusable workflow to indicate that it's not backward compatible and for renovate to refuse to propose an update to consume projects.

Perhaps this is implicitly built in by using major version tags, but it's not obvious and I haven't been able to spot anything in the docs.

**Comment by secustor** [comment]
If actions are pinned like that Renovate, will treat digest updates as the respective semver updates

https://github.com/renovatebot/renovate/blob/f8db9babf2944ee992a40965f89caa76e5ce42f7/.github/workflows/devcontainer.yml#L21

**Reply by electrofelix** [reply]
There's currently no semvers in the shared workflow repository in question because we haven't worked out how to deal with each tag applies to all workflows so major bumps indicating an incompatibility in one workflow will look like an incompatibility in all workflows even if non of the others even changed.

It's starting to seem like the only solution is a repo per reusable workflow if we want to handle incompatibilities on individual workflows.

**Reply by secustor** [reply]
If your repo does not uses only digests and no version there is no way to do this so far.
